metadata/ota_metadata/metadata_gen.py

```diff
@@ -16,10 +16,41 @@
 #!/usr/bin/env python3
 
 import os
-from hashlib import sha256
 import argparse
 import pathlib
+import zstandard
 import igittigitt
+from hashlib import sha256
+
+ZSTD_COMPRESSION_ALG = "zstd"
+ZSTD_COMPRESSION_LEVEL = 10
+ZSTD_MULTITHREADS = 2
+CHUNK_SIZE = 4 * (1024**2)  # 4MiB
+
+
+def zstd_compress_file(
+    cctx: zstandard.ZstdCompressor,
+    src_fpath: str,
+    dst_fpath: str,
+    *,
+    cmpr_ratio: float,
+    filesize_threshold: int,
+) -> bool:
+    if os.path.getsize(src_fpath) < filesize_threshold:
+        return False  # skip file with too small size
+
+    uncompressed_bytes, compressed_bytes = 0, 0
+    with open(src_fpath, "rb") as src_f, open(dst_fpath, "wb") as dst_f:
+        with cctx.stream_writer(dst_f) as compressor:
+            while data := src_f.read(CHUNK_SIZE):
+                uncompressed_bytes += len(data)
+                compressor.write(data)
+            compressed_bytes = compressor.tell()
+    # drop compressed file if cmpr ratio is too small
+    if uncompressed_bytes / compressed_bytes < cmpr_ratio:
```

Comment:
Prefer simpler to implementation.
```suggestion
    with open(src_fpath, "rb") as src_f, open(dst_fpath, "wb") as dst_f:
        with cctx.stream_writer(dst_f) as compressor:
            while data := src_f.read(CHUNK_SIZE):
                compressor.write(data)
    # drop compressed file if cmpr ratio is too small
    if os.path.getsize(src_fpath) / os.path.getsize(src_fpath) < cmpr_ratio:
```